File: lab2.py

# Task 1 

n = int(input("Please enter a positive integer greater than 9: "))

while n <= 9:
    n = int(input("Please enter a positive integer greater than 9: "))

current = n
steps = 0

print(current, end="")

while current >= 10:
    temp = current
    digit_sum = 0

    while temp > 0:
        digit_sum += temp % 10
        temp //= 10

    current = digit_sum
    steps += 1
    print(" →", current, end="")

print()
print("Final value:", current)
print("Total steps:", steps)


print("\n-------------------------------\n")


# Task 2 


n = int(input("Please enter a number between 10 and 100: "))

while n < 10 or n > 100:
    print("Invalid input. Please enter a number between 10 and 100:", end=" ")
    n = int(input())

fizz_count = 0
buzz_count = 0
fizzbuzz_count = 0

for i in range(1, n + 1):

    if i % 7 == 0:
        print(f"({i} is skipped)")
        continue

    if i % 3 == 0 and i % 5 == 0:
        print("FizzBuzz")
        fizzbuzz_count += 1

    elif i % 3 == 0:
        print("Fizz")
        fizz_count += 1

    elif i % 5 == 0:
        print("Buzz")
        buzz_count += 1

    else:
        print(i)

print("--- Summary ---")
print("Fizz count :", fizz_count)
print("Buzz count :", buzz_count)
print("FizzBuzz count:", fizzbuzz_count)


print("\n-------------------------------\n")



# Task 3


n = int(input("Please enter a number between 3 and 9: "))

while n < 3 or n > 9:
    n = int(input("Please enter a number between 3 and 9: "))

for i in range(1, 2 * n):

    if i <= n:
        k = i
    else:
        k = 2 * n - i

    for j in range(1, k + 1):
        print(j, end="")

    print()

File: lab2.cpp


#include <iostream>
using namespace std;

int main()
{

    
    // Task 1 
    

    int n;
    cout << "Please enter a positive integer greater than 9: ";
    cin >> n;

    while (n <= 9)
    {
        cout << "Please enter a positive integer greater than 9: ";
        cin >> n;
    }

    int current = n;
    int steps = 0;

    cout << current;

    while (current >= 10)
    {
        int temp = current;
        int digit_sum = 0;

        while (temp > 0)
        {
            digit_sum += temp % 10;
            temp /= 10;
        }

        current = digit_sum;
        steps++;

        cout << " -> " << current;
    }

    cout << endl;
    cout << "Final value: " << current << endl;
    cout << "Total steps: " << steps << endl;

    cout << "\n-------------------------------\n" << endl;


   
    // Task 2 
    

    cout << "Please enter a number between 10 and 100: ";
    cin >> n;

    while (n < 10 || n > 100)
    {
        cout << "Invalid input. Please enter a number between 10 and 100: ";
        cin >> n;
    }

    int fizz_count = 0;
    int buzz_count = 0;
    int fizzbuzz_count = 0;

    for (int i = 1; i <= n; i++)
    {

        if (i % 7 == 0)
        {
            cout << "(" << i << " is skipped)" << endl;
            continue;
        }

        if (i % 3 == 0 && i % 5 == 0)
        {
            cout << "FizzBuzz" << endl;
            fizzbuzz_count++;
        }
        else if (i % 3 == 0)
        {
            cout << "Fizz" << endl;
            fizz_count++;
        }
        else if (i % 5 == 0)
        {
            cout << "Buzz" << endl;
            buzz_count++;
        }
        else
        {
            cout << i << endl;
        }
    }

    cout << "--- Summary ---" << endl;
    cout << "Fizz count : " << fizz_count << endl;
    cout << "Buzz count : " << buzz_count << endl;
    cout << "FizzBuzz count: " << fizzbuzz_count << endl;

    cout << "\n-------------------------------\n" << endl;


    
    // Task 3 
   

    cout << "Please enter a number between 3 and 9: ";
    cin >> n;

    while (n < 3 || n > 9)
    {
        cout << "Please enter a number between 3 and 9: ";
        cin >> n;
    }

    for (int i = 1; i < 2 * n; i++)
    {
        int k;

        if (i <= n)
            k = i;
        else
            k = 2 * n - i;

        for (int j = 1; j <= k; j++)
        {
            cout << j;
        }

        cout << endl;
    }

    return 0;
}
